src/taskgraph.js

```diff
@@ -20,11 +20,42 @@ class TaskGraph {
     this.renderer = options.renderer || (process.stdout.isTTY ? new ConsoleRenderer() : new LogRenderer());
     this.locks = options.locks || {};
 
+    const provided = new Set();
+    this.nodes.forEach(({task: {provides}}) => provides.forEach(k => provided.add(k)));
     this.nodes.forEach(({task}) => {
       assert('title' in task, 'Task has no title');
       assert('run' in task, `Task ${task.title} has no run method`);
       assert(task.locks.every(l => l in this.locks), `Task ${task.title} has undefined locks`);
+      assert(task.requires.every(k => provided.has(k)), `Task ${task.title} has unfulfilled requirements`);
     });
+
+    if (options.target) {
+      const target = new Set(Array.isArray(options.target) ? options.target : [options.target]);
+      target.forEach(k => {
+        assert(provided.has(k), `Target ${k} is not provided by any nodes`);
+      });
+      this.nodes = this._target(target);
+    }
+  }
+
+  /**
+   * Reduce the set of nodes to just those required to provide `target` (a Set).
+   */
+  _target(target) {
```

Comment:
This bit feels like I'm back in school.

Reply:
Shades of @jonasfj (well, probably not -- it's a horribly inefficient algorithm!)